tests/unit_tests/corsika/test_corsika_output.py

```diff
@@ -1,29 +1,750 @@
+
+    # Repeat the test for less telescopes and see that less photons are counted in
+    corsika_output_instance.set_histograms(telescope_indices=[0, 1, 2])
+    hist_sums = [3677, 9.2, 3677, 3677]
+    for i_label, label in enumerate(["counts", "density", "direction", "time_altitude"]):
+        hist_values, x_edges, y_edges = corsika_output_instance._get_hist_2D_projection(label)
+        assert pytest.approx(np.sum(hist_values), 1e-2) == hist_sums[i_label]
+
+
+def test_get_2D_photon_position_distr(corsika_output_instance_set_histograms):
+    density = corsika_output_instance_set_histograms.get_2D_photon_position_distr(density=True)
+
+    # Test the values of the histogram
+    assert pytest.approx(np.sum(density[0]), 1e-2) == 29
+    counts = corsika_output_instance_set_histograms.get_2D_photon_position_distr(density=False)
+    assert pytest.approx(np.sum(counts[0]), 1e-2) == 11633
+
+    # The edges should be the same
+    assert (counts[1] == density[1]).all()
+    assert (counts[2] == density[2]).all()
+
+
+def test_get_2D_photon_direction_distr(corsika_output_instance_set_histograms):
+    for returned_variable in range(3):
+        assert (
+            corsika_output_instance_set_histograms.get_2D_photon_direction_distr()[
+                returned_variable
+            ]
+            == corsika_output_instance_set_histograms._get_hist_2D_projection("direction")[
+                returned_variable
+            ]
+        ).all()
+
+
+def test_get_2D_photon_time_altitude(corsika_output_instance_set_histograms):
+    for returned_variable in range(3):
+        assert (
+            corsika_output_instance_set_histograms.get_2D_photon_time_altitude()[returned_variable]
+            == corsika_output_instance_set_histograms._get_hist_2D_projection("time_altitude")[
+                returned_variable
+            ]
+        ).all()
+
+
+def test_get_2D_num_photons_distr(corsika_output_instance_set_histograms):
+    corsika_output_instance_set_histograms.telescope_indices = [0, 4, 10]
+    (
+        num_photons_per_event_per_telescope,
+        num_events_array,
+        telescope_indices_array,
+    ) = corsika_output_instance_set_histograms.get_2D_num_photons_distr()
+    assert len(num_events_array) == 3  # number of events in this output file + 1 (edges of hist)
+    assert (telescope_indices_array == [0, 1, 2, 3]).all()
+    assert (
+        pytest.approx(num_photons_per_event_per_telescope[0, 0], 1e-2) == 2543.3
+    )  # 1st tel, 1st event
+    assert (
+        pytest.approx(num_photons_per_event_per_telescope[0, 1], 1e-2) == 290.5
+    )  # 1st tel, 2nd event
+    assert (
+        pytest.approx(num_photons_per_event_per_telescope[1, 0], 1e-2) == 3054.1
+    )  # 2nd tel, 1st event
+    assert (
+        pytest.approx(num_photons_per_event_per_telescope[1, 1], 1e-2) == 128.81
+    )  # 2nd tel, 2nd event
+
+
+def test_get_hist_1D_projection(corsika_output_instance_set_histograms):
```

Comment:
May I suggest to move this test above all the others which call `_get_hist_1D_projection`? It would make more sense I think.